frontend/src/wizards/RoleAssignment/ProjectsTable.tsx

```diff
@@ -0,0 +1,230 @@
+          label: project.name,
+          value: project.name,
+        })),
+      },
+      {
+        id: 'type',
+        label: t('Type'),
+        tableFilterFn: (selectedValues, project) => selectedValues.includes(project.type),
+        options: [
+          {
+            label: 'Namespace',
+            value: 'Namespace',
+          },
+        ],
+      },
+      {
+        id: 'clusters',
+        label: t('Clusters'),
+        tableFilterFn: (selectedValues, project) =>
+          project.clusters.some((cluster) => selectedValues.includes(cluster)),
+        options: Array.from(new Set(projectsData.flatMap((project) => project.clusters))).map((cluster) => ({
+          label: cluster,
+          value: cluster,
+        })),
+      },
+    ]
+
+    return allFilters
+  }, [t, projectsData])
+
+  const columns: IAcmTableColumn<ProjectTableData>[] = [
+    {
+      header: t('Name'),
+      sort: 'name',
+      search: 'name',
+      cell: (project) => project.name,
+    },
+    {
+      header: t('Type'),
+      sort: 'type',
+      search: 'type',
+      cell: (project) => (
+        <Label isCompact color="blue">
+          {project.type}
+        </Label>
+      ),
+    },
+    {
+      header: t('Clusters'),
+      cell: (project) => {
+        return (
+          <LabelGroup numLabels={2}>
+            {project.clusters.map((cluster) => (
+              <Label key={cluster} isCompact color="grey">
+                <Link
+                  to={generatePath(NavigationPath.clusterOverview, {
+                    namespace: cluster,
+                    name: cluster,
+                  })}
+                  style={{ textDecoration: 'none' }}
+                >
+                  {cluster}
+                </Link>
+              </Label>
+            ))}
+          </LabelGroup>
+        )
+      },
+    },
+  ]
+
+  const handleSelect = useCallback(
+    (projects: ProjectTableData[]) => {
+      setHasSelectedProjects(projects.length > 0)
+      if (onSelectionChange) {
+        onSelectionChange(projects)
+      }
+    },
+    [onSelectionChange]
+  )
+
+  return (
+    <AcmTable<ProjectTableData>
+      items={isLoading ? undefined : projectsData}
+      columns={columns}
+      keyFn={(project) => `${project.name}-${project.clusters.join(',')}`}
+      tableActionButtons={tableActionButtons.length > 0 ? tableActionButtons : undefined}
+      onSelect={handleSelect}
+      filters={filters}
+      searchPlaceholder={t('Search projects')}
+      autoHidePagination
+      emptyState={
+        <AcmEmptyState
+          key="projectsEmptyState"
+          title={t('No common projects found')}
+          message={t('Go back and select different clusters, or create projects with the same name on these clusters.')}
+          action={
+            onCreateClick ? (
+              <AcmButton variant="primary" onClick={onCreateClick}>
+                {t('Create common project')}
+              </AcmButton>
+            ) : undefined
```

Comment:
isn't `null` a more appropriated value? (see https://medium.com/@o_o_o/null-vs-undefined-can-i-use-only-one-a3b7db5468f2)
```suggestion
            ) : null
```